libs/ui/src/hooks/use-activity.hook.ts

```diff
@@ -31,50 +32,109 @@ const transformApiData = (
-  );
 
   let sectionListItem: SectionListActivityData | undefined;
 
   const userTokensMetadata: ActivityResponse['token_dict'] = data.token_dict;
+  const userProjectsMetadata: ActivityResponse['project_dict'] = data.project_dict;
 
-  filteredTransactions.forEach(txData => {
+  const getTokenSymbol = (address: string) =>
+    userTokensMetadata[address].optimized_symbol ?? userTokensMetadata[address].name;
+
+  const getProjectName = (projectId: string) => userProjectsMetadata[projectId].name;
+
+  const getTransactionType = (txData: TransactionResponse, isGasToken: boolean, isGasTokenSent: boolean) => {
+    if (txData.cate_id === TransactionTypeEnum.Send || (isGasToken && isGasTokenSent)) {
+      return {
+        type: TransactionTypeEnum.Send,
+        label: TransactionLabelEnum.Send
+      };
+    }
+    if (txData.cate_id === TransactionTypeEnum.Receive || (isGasToken && !isGasTokenSent)) {
+      return {
+        type: TransactionTypeEnum.Receive,
+        label: TransactionLabelEnum.Received
+      };
+    }
+    if (isNotEmptyString(txData.tx.name)) {
+      return {
+        type: TransactionTypeEnum.Interaction,
+        label: txData.tx.name
+      };
+    }
+
+    return {
+      type: TransactionTypeEnum.ContractInteraction,
+      label: TransactionLabelEnum.ContractInteraction
+    };
+  };
+
+  data?.history_list.forEach(txData => {
     const activityData = {
-      transactionStatus: TransactionStatusEnum.applied,
+      status: TransactionStatusEnum.applied,
       hash: txData.id,
       timestamp: txData.time_at
     } as ActivityData;
 
     if (txData.tx?.status === 0) {
-      activityData.transactionStatus = TransactionStatusEnum.failed;
-    } else {
-      activityData.transactionStatus = TransactionStatusEnum.applied;
+      activityData.status = TransactionStatusEnum.failed;
     }
 
-    const getTokenSymbol = (address: string) =>
-      userTokensMetadata[address].optimized_symbol ?? userTokensMetadata[address].name;
-
-    if (txData.cate_id !== null) {
-      activityData.transactionLabel = capitalize(txData.cate_id) as TransactionLabelEnum;
-
-      if (activityData.transactionLabel === TransactionLabelEnum.Send) {
-        activityData.amount = txData.sends[0]?.amount;
-        activityData.tokenId = txData.sends[0]?.token_id;
-        activityData.symbol = getTokenSymbol(txData.sends[0]?.token_id);
-      } else {
-        activityData.tokenId = txData.receives[0]?.token_id;
-        activityData.amount = txData.receives[0]?.amount;
-        activityData.symbol = getTokenSymbol(txData.receives[0]?.token_id);
-        activityData.transactionLabel = TransactionLabelEnum.Received;
+    const isGasToken =
+      !(txData.sends.length > 0 && txData.receives.length > 0) &&
+      isDefined(txData.tx) &&
+      txData.tx.value > 0 &&
+      txData.cate_id === null &&
+      txData.token_approve === null;
+    const isGasTokenSent = publicKeyHash.toLowerCase() === txData.tx?.from_addr?.toLowerCase();
```

Comment:
create separate util functions `isGasTokenTransaction` and `isGasTokenSentTransaction`